firmware/controllers/engine_controller_misc.cpp

```diff
@@ -67,7 +67,8 @@ volatile uint32_t lastLowerNt = 0;
 volatile uint32_t upperTimeNt = 0;
 
 efitick_t getTimeNowNt() {
-	chibios_rt::CriticalSectionLocker csl;
+	// TODO: lock free so we don't have to disable interrupts!
+	__disable_irq();
```

Comment:
May want to make this safe in case interrupts are already disabled when this function is called.  Or maybe I should go commit my lockfree version.

Reply:
+1 vote for lock free version :heart: